cmd/gctcli/data_history.go

```diff
@@ -139,34 +63,228 @@ var dataHistoryCommands = &cli.Command{
 			ArgsUsage: "<nickname>",
 			Action:    getDataHistoryJobSummary,
 			Flags: []cli.Flag{
-				&cli.StringFlag{
-					Name:  "nickname",
-					Usage: "binance-spot-btc-usdt-2019-trades",
-				},
+				nicknameFlag,
 			},
 		},
+		dataHistoryJobCommands,
 		{
-			Name:   "addnewjob",
-			Usage:  "creates a new data history job",
-			Flags:  fullJobSubCommands,
-			Action: upsertDataHistoryJob,
+			Name:      "deletejob",
+			Usage:     "sets a jobs status to deleted so it no longer is processed",
+			ArgsUsage: "<id> or <nickname>",
+			Flags:     specificJobSubCommands,
+			Action:    setDataHistoryJobStatus,
 		},
 		{
-			Name:   "upsertjob",
-			Usage:  "adds a new job, or updates an existing one if it matches jobid OR nickname",
-			Flags:  fullJobSubCommands,
-			Action: upsertDataHistoryJob,
+			Name:      "pausejob",
+			Usage:     "sets a jobs status to paused so it no longer is processed",
+			ArgsUsage: "<id> or <nickname>",
+			Flags:     specificJobSubCommands,
+			Action:    setDataHistoryJobStatus,
 		},
 		{
-			Name:      "deletejob",
-			Usage:     "sets a jobs status to deleted so it no longer is processed",
+			Name:      "unpausejob",
+			Usage:     "sets a jobs status to active so it can be processed",
 			ArgsUsage: "<id> or <nickname>",
 			Flags:     specificJobSubCommands,
-			Action:    deleteDataHistoryJob,
+			Action:    setDataHistoryJobStatus,
+		},
+		{
+			Name:      "updateprerequisite",
+			Usage:     "adds or updates a prerequisite job to the job referenced - if the job is active, it will be set as 'paused'",
+			ArgsUsage: "<prerequisite> <nickname>",
+			Flags:     prerequisiteJobSubCommands,
+			Action:    setPrerequisiteJob,
+		},
+		{
+			Name:      "removeprerequisite",
+			Usage:     "removes a prerequisite job from the job referenced - if the job is 'paused', it will be set as 'active'",
+			ArgsUsage: "<nickname>",
+			Flags: []cli.Flag{
+				nicknameFlag,
+			},
+			Action: setPrerequisiteJob,
+		},
+	},
+}
+
+var dataHistoryJobCommands = &cli.Command{
+	Name:      "addjob",
+	Usage:     "add or update data history jobs",
+	ArgsUsage: "<command> <args>",
+	Subcommands: []*cli.Command{
+		{
+			Name:   "savecandles",
+			Usage:  "will fetch candle data from an exchange and save it to the database",
+			Flags:  append(baseJobSubCommands, dataHandlingJobSubCommands...),
+			Action: upsertDataHistoryJob,
+		},
+		{
+			Name:   "convertcandles",
+			Usage:  "convert candles saved to the database to a new resolution eg 1min -> 5min",
+			Flags:  append(baseJobSubCommands, candleConvertJobJobSubCommands...),
+			Action: upsertDataHistoryJob,
+		},
+		{
+			Name:   "savetrades",
+			Usage:  "will fetch trade data from an exchange and save it to the database",
+			Flags:  append(baseJobSubCommands, dataHandlingJobSubCommands...),
+			Action: upsertDataHistoryJob,
+		},
+		{
+			Name:   "converttrades",
```

Comment:
Wondering if these flags are needed in this context?
```json
--request_size_limit value  the number of candles to retrieve per API request (default: 500)
   --max_retry_attempts value  the maximum retry attempts for an interval period before giving up (default: 3)
   --batch_size value          the amount of API calls to make per run (default: 3)
   --overwrite_existing_data   will process and overwrite data if matching data exists at an interval period. if false, will not process or save data (default: false)
```

Reply:
Yes they do. I can update the descriptions to be more relevant, but they all use the same parameters.
- request size limit helps calculate the data ranges to fetch
- max retry attempts is still valid as when you create a job, there's no guarantee the data is already there, and a job may be running at the same time to save said data its trying to convert - that onus is on the user.
- batch size dictates how many cycles per processing of a job to run. We don't convert all trades in one go to be considerate of resources
- overwrite existing data is important as there may already be saved data that differs at the destination

Reply:
Okay makes sense now! 🚀